tools/tool-testing/galaxy-utils.js

```diff
@@ -0,0 +1,251 @@
+    var allSettings = _.extend(galaxySettings, settings);
+    var settingsFile = "settings-" + appName + ".json";
+    sandbox.write(settingsFile, JSON.stringify(allSettings));
+
+    testUtils.createAndDeployApp(sandbox, {
+      settingsFile: "../" + settingsFile,
+      appName: appName,
+      templateApp: templateApp
+    });
+  } else {
+    testUtils.createAndDeployApp(sandbox, {
+      appName: appName,
+      templateApp: templateApp
+    });
+  }
+
+  // Galaxy might take a while to spin up an app.
+  exports.waitForContainers();
+
+  return appName + "." + GALAXY_URL;
+
+});
+
+// Cleanup the app by deleting it from Galaxy.
+//
+// XXX: We should also clean out its Mongo, but we don't, since, currently, none
+// of our apps actually put any records into it.
+exports.cleanUpApp = selftest.markStack(function (sandbox, appName) {
+  testUtils.cleanUpApp(sandbox, appName);
+
+  // Galaxy might take a while to spin up an app, though it should be fairly
+  // quick.
+  exports.waitForContainers();
+});
+
+//////////////////////////////////////////////////////////////////////////////
+//  We want to test some of the server-side functionality that doesn't actually
+//  have a command-line API right now. Below functionas are going to use the
+//  tool's ability to make a DDP client and connect to the server to call
+//  methods directly.
+////////////////////////////////////////////////////////////////////////////////
+
+// Returns a logged in connection to GalaxyAPI
+exports.loggedInGalaxyAPIConnection = selftest.markStack(function () {
+  // The credentials of the user might not be the credentials of the galaxytester.
+  auth.doInteractivePasswordLogin({
+    username: GALAXY_USERNAME,
+    password: GALAXY_PASSWORD
+  });
+  var galaxyDomain = GALAXY_URL;
+  var galaxyUrl = "https://" + galaxyDomain;
+  return authClient.loggedInConnection(
+    galaxyUrl,
+    galaxyDomain,
+    "galaxy-api"
+  );
+});
+
+// If the connection has disconnected, close it and open a new one. (workaround
+// for the fact that connections in the tool do not reconnect)
+exports.renewConnection = selftest.markStack(function (conn) {
+  if (!conn.connected) {
+    conn.close();
+    conn = exports.loggedInGalaxyAPIConnection();
+  }
+  return conn;
+});
+
+// Given a connection, makes a call to Galaxy API.
+exports.callGalaxyAPI = function (conn, ...args) {
+  conn = exports.renewConnection(conn);
+  return conn.call(...args);
+};
+
+// Gets app record from Galaxy API by name.
+//
+// This method will create and manage its own connection.
+exports.getAppRecordByName = selftest.markStack(function (appName) {
+  var conn = exports.loggedInGalaxyAPIConnection();
+  var appRecord = {};
+  conn.connection.registerStore('apps', {
+    update: function (msg) {
+      if (msg.msg === 'added' && msg.fields &&
+          msg.fields.hostname === appName) {
+        appRecord = _.extend({ _id: msg.id }, msg.fields);
+      }
+    }
+  });
+  conn.subscribeAndWait("apps");
```

Comment:
Oh I like literally just deleted it. Perhaps this is the problem I'm seeing? 

It should be `("/user/apps", orgName)`